src/savegame.cpp

```diff
@@ -77,6 +77,9 @@ void game::serialize( std::ostream &fout )
             json.member( "last_target", critter_tracker->temporary_id( *mon ) );
             json.member( "last_target_type", -1 );
         }
+    } else {
+        tripoint templtp = u.last_target_pos ? *u.last_target_pos : tripoint_min;
+        json.member( "last_target_pos", templtp );
```

Comment:
You can serialize a `cata::optional` directly, no need to convert it into a `tripoint` and serialize that. Same for deserialization.

Reply:
I tried to do:
serialize: ```json.member( "last_target_pos", *u.last_target_pos );```
deserialize: ```data.read( "last_target_pos", *u.last_target_pos );```

But this gives me an error on save: ```cataclysm: src/optional.h:41: T &cata:optional<tripoint>::get() [T = tripoint]: Assertion 'full' failed.```

Reply:
u.last_target_pos is cata::optional. Use it without a *.

Reply:
Tried that, but then this happens: (same with data.read)
```src/json.h:583:13: error: no matching member function for call to 'write'
            write( value );
            ^~~~~
src/savegame.cpp:81:14: note: in instantiation of function template specialization 'JsonOut::member<cata::optional<tripoint> >' requested here
        json.member( "last_target_pos", u.last_target_pos );
             ^
```

Reply:
@BevapDin Alright found out whats going on here.
You put the de/serialize functions into savegame_json.cpp
I need them in savegame.cpp though. Where'd I best put them
to access from both files without adding any more includes to .h files?

Reply:
Move those definitions into "enums.h" and make them `inline` (and add a forward declaration of `cata::optional` there as well).